### ISSUE TITLE
Feature request: unpouse to rewind

### ISSUE THREAD

**Issue by thekomer** [flackbash/AudioAnchor#22]
Hey! Nice app!

There's something, maybe useful, that you could implement.

It would be nice if app would have automatically rewind few seconds when I resume playback. (The number of seconds previously set by user; 0 as default).

Reasoning: I'm just listening to a book. And I find myself pushing pouse very frequently (I need to translate some words i don't understand) and when i resume, all I need is few last seconds to catch the thread of thoughts. I know there is rewind button in notification for 30 sec rewind, but that's to much. Even 10 sec is to much if I pouse the book frequently.

Specially, If I was listening to a book while my phone was in the pocket and I play/pouse with the remote control on my headphones. Then, rewind button is out of the reach. 

:) Tnx

**Comment by flackbash** [comment]
I like the idea if we add it as an option in the settings with 0 as default.

@axoTheLotl: Do you want to give this a try? It should not require a lot of changes and the changes are probably restricted to the MediaPlayerService and files connected to the settings. Let me know if you want to implement this :)

**Comment by Sulter** [comment]
I'm definitely all in for this feature too. And as @flackbash said, just have it adjustable - like the "Fade out time" option.

**Comment by thekomer** [comment]
Hey there again! I'm glad you like my idea.

Of course I meant the feature to be optional (set to zero by default) and configurable (yes, something like fadeout time).

In the meanwhile I discovered one more thing.  
When I play a book, pouse and unpouse... the part of the sentence is skipped. I didn't realise this before because I would automatically push rewind button after unpouse.
I tested it.  It skipped the part of a sentence that goes like this: "...that to the competition, and let..." or "...everything that could be possibly...". I don't know why is that,  It's like a second or maybe two seconds skipped. Maybe it's just on my device, or it's just the way media services work. You can test it on your device if that's something that interests you.

From that new perspective, I think "rewind on unpouse" feature is even more logical. But I let the decision to you :-)

Thank you again for considering my idea.

C ya!

**Comment by axoTheLotl** [comment]
I'd love to implement this! It's definitely a useful feature and it shouldn't be too hard to do. Thanks for asking! : )
End of the week I can get around to it.

As for the skipping:
Maybe have a look at this issue: https://github.com/flackbash/AudioAnchor/issues/17 
Sounds like they could be connected.

I learned that the problem was with the files. So maybe test with different files from other sources. 
If the files are the problem you can maybe put them through an audiofile converter. That solved it for me.

**Comment by flackbash** [comment]
> As for the skipping:
Maybe have a look at this issue: #17

Exactly. Especially look out for files with variable bit rate. If they are several hours long, they almost always cause issues like the one you encountered (not only in AudioAnchor, but also in VLC and most other players). I often had problems when using ffmpeg for the conversion of files. Under Linux you can use [SoundConverter](https://soundconverter.org/) instead and specify a constant bitrate. That should do the trick.

If that solves the problem or something else does, please let us know. Would be interesting to have some more insights on this.

**Comment by thekomer** [comment]
Hey!  Regarding skipping  issue....
I tried many times, and other files from the same book. It skips. Sometimes backwards, bit mostly forward.

Unfortunately i dont have different book to try it with.

But I tried the same file with different players. They don't have the same issue. Everything works normally. One is Musicolet (from Play store, not FOSS but free and adfree)), another is Mpv player from F-droid repos. This one is opensource as I know of.

My book is 64 kbps, 22050 Hz, 16 bps, mp3, mpeg-2 layer 3, mono, 25.63 MB,  55:59 min.

I will try to exepriment with converters, but right now I don't have time. Sorry. Next week maybe. And then I will download more books to experiment with them. 

I have found one more issue. But I think this one needs separate thread. 

C u there! :)
